doc/misc.mdz

```diff
@@ -60,3 +60,325 @@ x       y
   banana  orange
      cat     dog
 ```
+
+## map-keys
+
+Returns new table with function applied to table's keys recursively.
+
+@codeblock[janet]```
+(misc/map-keys string {1 2 3 4}) 
+# => @{"1" 2 "3" 4}
+```
+
+## map-vals
+
+Returns new table with function applied to table's values.
+
+@codeblock[janet]```
+(misc/map-vals string {1 2 3 4}) 
+# => @{1 "2" 3 "4"}
+```
+
+## select-keys
+
+Returns new table with selected keys from dictionary.
+
+@codeblock[janet]```
+(misc/select-keys {1 2 3 4 5 6} [1 5]) 
+# => @{1 2 5 6}
+```
+
+## cond->
+
+Threading conditional macro. It takes value to mutate, and clauses pairs with a
+condition and a operation to which the value is put as the first argument. All
+conditions are tried and for truthy conditions the operation is performed.
+
+Returns the value mutated if any condition is truthy.
+
+@codeblock[janet]```
+(misc/cond-> @{:a :b}
+                 (pos? 1) (put :a :c)
+                 (pos? 0) (put :a :d))
+# => @{:a :c}
+```
+
+## cond->>
+
+Threading conditional macro. It takes value to mutate, and clauses pairs of a
+condition and a operation to which the value, is put as the last argument. All
+conditions are tried and for truthy the operation is performed.
+
+Returns mutated value if any condition is truthy.
+
+@codeblock[janet]```
+(misc/cond->> @{:a :b}
+              (pos? 1) (merge {:d :e})
+              (pos? 0) (merge {:e :f}))
+# => @{:a :b :d :e}
+```
+
+## make
+
+Creates new table from even number of key value pairs in a variadic arguments,
+or take existing table if it is one argument. Then it sets tables' prototype
+to the prototype argument.
+
+Factory function for creating new objects from prototypes.
+
+@codeblock[janet]```
+(def Proto @{:greet |(string "Hello " ($ :name))})
+(def t (misc/make Proto :name "pepe"))
+(:greet t)
+# => "Hello pepe"
+```
+
+## do-var
+
+Convenience macro for defining varible with value before body and returning
+it after evaluating body, that presumably modifies variable.
+
+@codeblock[janet]```
+(misc/do-var res 0 (set res 100) (++ res))
+# => 101
+```
+
+## do-def
+
+Convenience macro for defining constant with value before body and returning it
+after evaluating body, that presumably modifies the const refered content. For
+example buffer, table or array.
+
+@codeblock[janet]```
+(misc/do-def res @"" (buffer/push res "a")) 
+# => @"a"
+```
+
+## capout
+
+Captures the standart output of the variadic body and returns it as a buffer.
+
+@codeblock[janet]```
+(misc/capout 
+  (def p "HOHOHO")
+  (prin p))
+# => @"HOHOHO"
+```
+
+## caperr
+
+Captures the standart error output of the variadic body and returns it as
+a buffer.
+
+@codeblock[janet]```
+(misc/capout
+  (def p "HOHOHO")
+  (prin p))
```

Comment:
needs edit after copy&paste

Reply:
I have looked at this part and do not see any problem. May you please be more specific? I found another problem with a block on line 187 though :-D.

Reply:
it is the example from capout copy&pasted to caperr and not:
```
(misc/caperr
  (def p "HOHOHO")
  (eprin p))
```

Reply:
Oh, the programmer's blindness. Thank you very much!